templates/cisco_nxos_show_version.textfsm

```diff
@@ -1,18 +1,71 @@
-Value UPTIME ((\d+\s\w+.s.,?\s?){4})
+Value UPTIME ((\d+\s\w+(\(s\))?(,\s)?){4,5})
 Value LAST_REBOOT_REASON (.+)
-Value OS (\d+.\d+(.+)?)
+Value OS (\d+\..*)
 Value BOOT_IMAGE (.*)
-Value PLATFORM (\w+)
+Value PLATFORM ([\w-]+)
 Value HOSTNAME (.*)
+Value SERIAL (\w+)
+
 
 Start
-  ^\s+(NXOS: version|system:\s+version)\s+${OS}\s*$$
+  ^\s*(NXOS:|system:|Cisco IOS XE Software,)\s+[Vv]ersion:?\s+${OS}$$
+  # Skip duplicate half version
+  ^\s+System version:
   ^\s+(NXOS|kickstart)\s+image\s+file\s+is:\s+${BOOT_IMAGE}\s*$$
   ^\s+cisco\s+${PLATFORM}\s+[cC]hassis
   ^\s+cisco\s+Nexus\d+\s+${PLATFORM}
   # Cisco N5K platform
   ^\s+cisco\s+Nexus\s+${PLATFORM}\s+[cC]hassis
   ^\s+Device\s+name:\s+${HOSTNAME}$$
   ^\s+cisco\s+.+-${PLATFORM}\s*
-  ^Kernel\s+uptime\s+is\s+${UPTIME}
-  ^\s+Reason:\s${LAST_REBOOT_REASON} -> Record
+  # Nexus intel platform uses Board ID as serial/license
+  ^\s*Processor\s[Bb]oard\sID\s+${SERIAL}$$
```

Comment:
It looks like this is attempting to be used as Cat9K, but these are managed via different templates.

Please remove any IOS specific things as well as the new test file that is IOS-XE related.

I would suggest just capturing the serial number and not adding the `. ^. -> Error` since there are plenty of lines we don't care about and may make maintaining this template harder than other templates where most of the data needs to be captured.

Reply:
Thanks for the review and feedback, trying to implement/simplify/fix it now.